nmrcraft/models/models.py

```diff
@@ -1,8 +1,42 @@
+import inspect
+
 from sklearn.ensemble import GradientBoostingClassifier, RandomForestClassifier
 from sklearn.linear_model import LogisticRegression
 from sklearn.svm import SVC
 
 
+class InvalidModelNameError(ValueError):
+    def __init__(self, model_name, models):
+        super().__init__(f"Model {model_name} not found. Available models are {list(models.keys())}")
+
+
+class InvalidArgumentError(ValueError):
+    def __init__(self, kwarg, model_name):
+        super().__init__(f"Invalid argument {kwarg} for model {model_name}")
+
+
+class InvalidModelNameTypeError(TypeError):
+    def __init__(self):
+        super().__init__("model_name must be a string")
+
+
+def validate_model_instance(model_name):
+    if not isinstance(model_name, str):
+        raise InvalidModelNameTypeError()
+
+
+def validate_model_availability(model_name, models):
+    if model_name.lower() not in models:
+        raise InvalidModelNameError(model_name, models)
+
+
+def validate_kwargs(kwargs, model_class, model_name):
+    args = inspect.signature(model_class.__init__).parameters.keys()
+    for kwarg in kwargs:
+        if kwarg not in args:
+            raise InvalidArgumentError(kwarg, model_name)
+
+
 def load_model(model_name, **kwargs):
```

Comment:
Let’s add a trype hint here, which makes the “InvalidModelNameError” obsolete. By adding a type hint, you atomatically force the model_name to be a string:

def load_model(model_name:str, **kwargs):